tests/assembly_02.cc

```diff
@@ -121,18 +219,120 @@ main(int argc, char **argv)
       {
         patch.reinit(cell, n_overlap);
 
-        const auto                           n_dofs_patch = patch.n_dofs();
+        double H = 1.0 / n_subdivisions;
+        double h = H / fe_degree;
+
+        const auto                           n_dofs_patch  = patch.n_dofs();
+        const unsigned int                   N_dofs_coarse = patch.n_cells();
+        const unsigned int                   N_dofs_fine   = n_dofs_patch;
         std::vector<types::global_dof_index> local_dof_indices_fine(
           n_dofs_patch);
         patch.get_dof_indices(local_dof_indices_fine);
 
         AffineConstraints<double> patch_constraints;
         for (unsigned int d = 0; d < 2 * dim; ++d)
-          patch.make_zero_boundary_constraints<double>(d, patch_constraints);
+          if (patch.at_boundary(d))
+            patch.make_zero_boundary_constraints<double>(d, patch_constraints);
+          else
+            patch.make_zero_boundary_constraints<double>(d, patch_constraints);
```

Comment:
```suggestion
          if (patch.at_boundary(d))
            patch.make_zero_boundary_constraints<double>(d, patch_constraints);
          else
            patch.make_zero_boundary_constraints<double>(d, patch_constraints);
```

@camillabelponer The first gives you the boundary dofs, the second ones the internal-boundary dofs (one needs to remove the duplicate ones from the actual boundary dofs). The rest [0, N_dofs_fine[ is internal.